packages/flutter/test/material/checkbox_test.dart

```diff
@@ -185,4 +186,43 @@ void main() {
     await tester.pumpAndSettle();
     expect(checkBoxValue, null);
   });
+
+  testWidgets('has semantic events', (WidgetTester tester) async {
+    dynamic semanticEvent;
+    bool checkboxValue = false;
+    SystemChannels.accessibility.setMockMessageHandler((dynamic message) {
+      semanticEvent = message;
+    });
+    final SemanticsTester semanticsTester = new SemanticsTester(tester);
+
+    await tester.pumpWidget(
+      new Material(
+        child: new StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return new Checkbox(
+              value: checkboxValue,
+              onChanged: (bool value) {
+                setState(() {
+                  checkboxValue = value;
+                });
+              },
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.byType(Checkbox));
+    final RenderObject object = tester.firstRenderObject(find.byType(Checkbox));
+
+    expect(checkboxValue, true);
+    expect(semanticEvent, <String, dynamic>{
+      'type': 'tap',
+      'nodeId': object.debugSemantics.id,
+      'data': <String, dynamic>{},
+    });
```

Comment:
Bonus points for asserting that the node with id `object.debugSemantics.id` was actually annotated with the click action for the checkbox in the semantics tree.

Same might be useful for some of the other tests.

Reply:
Done for all tests except tooltip